src/hash_table.py

```
@@ -0,0 +1,68 @@
+"""Module for hash_table.py."""
+
+
+class HashTable:
+    """Create hash table object."""
+
+    entriesCount = 0
+    slots = []
+
+    def __init__(self, num_slots=1031, hash_function=0):
+        """Init declaration."""
+        self.num_slots = num_slots
+        self.slots = [[] for i in range(num_slots)]
+        if hash_function:
+            self._hash_function = self._complex_hash
+        else:
+            self._hash_function = self._naive_hash
+
+    def get(self, key):
+        """Return the value stored with given key."""
+        if not isinstance(key, str):
+            raise KeyError("key is not string.")
+        correct_bucket = self._hash(key) % self.num_slots
+        for it in self.slots[correct_bucket]:
+            if it[0] == key:
+                return it[1]
+        print("key is not in table.")
+        return None
```

A hash table should raise an appropriate error when you try to 'get' a key not in the table.